### DISCUSSION TITLE
Scroll Past End of SceneTree

### DISCUSSION THREAD

**Discussion by Jay-Dorcat** [godotengine/godot-proposals#8557] [Editor]
### Describe the problem or limitation you are having in your project
- This is less of an issue/problem/limitation and more something that makes working on scenes with many objects feel a bit nicer.
### Describe the feature / enhancement and how it helps to overcome the problem or limitation:
- An Editor option to scroll past the end of the SceneTree the same way you can Scroll past End of File in the text editor so that scrolling down will always reveal empty space below your existing nodes.
- Being able to scroll past end stops the workspace from feeling cramped and that there is still more room for nodes, otherwise it can feel like you're running out of space.
### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams:
- I wouldn't know how best to do it.
- It might be a better option to add the Scroll past End functionality to scroll containers, and then add the editor option to toggle it.
### If this enhancement will not be used often, can it be worked around with a few lines of script?:
- It changes the editor so would need to be an addon or core change
### Is there a reason why this should be core and not an add-on in the asset library?:
- It might be able to be done as an addon but it is such a small change and would be wanted on almost all projects.

**Comment by Calinou** [comment]
This needs to be implemented in the Tree node itself, so the scene tree dock can make use of this. That said, I've rarely ever seen this UI pattern in anything else than code editors.
